### PR TITLE
Show exception

### PR DESCRIPTION
**Pull request by devoncarew**
- when stopping on an exception, display the toString() of an exception in the console (fix https://github.com/flutter/flutter/issues/4037)
- when stopping on an exception, display the exception object in the stack frame view of the first frame
- fix an npe when hitting the resume button (fix https://github.com/dart-atom/dartlang/issues/996)

@danrubel 

screen shot 2016-05-19 at 10 51 46 pm

screen shot 2016-05-20 at 8 01 48 am

### CODE REVIEW

**Comment by devoncarew** [comment]
(also, change the break-on-exceptions default to `uncaught` from `all`. We were too aggressive about stopping on flutter exceptions: https://github.com/flutter/flutter/issues/3899. We'll need to coordinate with the flutter framework to show exception locations for handled exceptions)

File: lib/debug/observatory_debugger.dart [hunk 1]
```diff
@@ -398,8 +395,38 @@ class ObservatoryConnection extends DebugConnection {
...
+    return ExceptionPauseMode.kUnhandled;
+  }
+
+  void _printExceptionToConsole(ObservatoryIsolate isolate, InstanceRef exception) {
+    if (exception.kind == InstanceKind.kString) {
+      launch.pipeStdio(
+        "exception: '${exception.valueAsString}'\n",
+        error: true
+      );
+    } else if (exception.valueAsString != null) {
```
Comment (danrubel) [inline lib/debug/observatory_debugger.dart]: Combine if(A) and else if(B) into if(A||B) because blocks are identical ?

**Comment by danrubel** [comment]
LGTM

File: lib/debug/observatory_debugger.dart (hunk 1 above)
Comment (devoncarew) [inline lib/debug/observatory_debugger.dart]: will do

**Pull request merged**